packages/expo-cli/src/credentials/views/IosDistCert.ts

```diff
@@ -360,6 +363,13 @@ async function selectDistCertFromList(
     return null;
   }
 
+  if (ctx.nonInteractive) {
+    throw new CommandError(
+      'NON_INTERACTIVE',
+      "Start the CLI without the '--non-interactive' flag to select a certificate."
+    );
+  }
+
```

Comment:
```suggestion
```
If you have seen that case to happen somewhere in nonInteractive mode it should be handled earlier